src/firebolt/common/util.py

```diff
@@ -37,15 +51,52 @@ def fix_url_schema(url: str) -> str:
     return url if url.startswith("http") else f"https://{url}"
 
 
-def async_to_sync(f: Callable) -> Callable:
+class AsyncJobThread:
+    def __init__(self) -> None:
+        self.loop: Optional[AbstractEventLoop] = None
+        self.result: Optional[Any] = None
+        self.exception: Optional[BaseException] = None
+        super().__init__()
+
+    def _initialize_loop(self) -> None:
+        if not self.loop:
+            try:
+                self.loop = get_event_loop()
+            except RuntimeError:
+                self.loop = new_event_loop()
```

Comment:
I think part of the functionality of `get_event_loop` is that it will create a loop if one is not running. Would there be a case when `except:` here happens?

Reply:
This happens almost every time. Despite the description, it fails with runtime if no event loop is set

Reply:
Interesting. Let's pop a quick comment there saying this happens despite get_event_loop claiming to do otherwise.